python2/httplib2/__init__.py

```diff
@@ -2133,13 +2135,19 @@ def request(
                         cachekey,
                     )
         except Exception as e:
+            isTimeout = isinstance(e, socket.timeout)
+            if isTimeout:
+                conn = self.connections.pop(conn_key, None)
+                if conn:
+                    conn.close()
+                    
             if self.force_exception_to_status_code:
                 if isinstance(e, HttpLib2ErrorWithResponse):
                     response = e.response
                     content = e.content
                     response.status = 500
                     response.reason = str(e)
-                elif isinstance(e, socket.timeout):
+                elif isTimeout:
```

Comment:
```suggestion
                elif is_timeout:
```

Surprised to say it, but Github is getting better each day after that MS deal.